src/main/java/crazypants/enderio/machine/reservoir/BlockReservoir.java

```diff
@@ -69,11 +70,18 @@ public boolean onBlockActivated(World world, int x, int y, int z, EntityPlayer e
         }
         return true;
       }
+      if (FluidUtil.fillInternalTankFromPlayerHandItem(world, x, y, z, entityPlayer, tank)) {
```

Comment:
it does not contain the getController() code. So it can only fix the internal tanks if the player clicks the controller block by chance. And, because you present the tanks in the wrong order, in the other 3 cases, no water regens.